src/Microsoft.AspNet.Identity/SignInManager.cs

```diff
@@ -53,24 +56,17 @@ public class SignInManager<TUser> where TUser : class
         internal IUserClaimsPrincipalFactory<TUser> ClaimsFactory { get; private set; }
         internal IdentityOptions Options { get; private set; }
         internal ILogger Logger { get; set; }
+        protected IdentityLogger IdentityLogger { get; set; }
 
         // Should this be a func?
-        public virtual async Task<ClaimsPrincipal> CreateUserPrincipalAsync(TUser user)
-        {
-            return await ClaimsFactory.CreateAsync(user);
-        }
+        public virtual async Task<ClaimsPrincipal> CreateUserPrincipalAsync(TUser user) => await ClaimsFactory.CreateAsync(user);
 
         public virtual async Task<bool> CanSignInAsync(TUser user)
         {
-            if (Options.SignIn.RequireConfirmedEmail && !(await UserManager.IsEmailConfirmedAsync(user)))
-            {
-                return await LogResultAsync(false, user);
-            }
-            if (Options.SignIn.RequireConfirmedPhoneNumber && !(await UserManager.IsPhoneNumberConfirmedAsync(user)))
-            {
-                return await LogResultAsync(false, user);
-            }
-            return await LogResultAsync(true, user);
+            var result = !(Options.SignIn.RequireConfirmedEmail && !(await UserManager.IsEmailConfirmedAsync(user))) ||
```

Comment:
This is a pretty gross looking line of code, personally I'm not a fan of compacting things to the point where its impossible to read due to 4 !'s an or, and 2 &'s